driver/src/contract.rs

```diff
@@ -327,13 +332,41 @@ impl SnappContract for SnappContractImpl {
         self.contract
             .call(
                 "applyAuction",
+                (slot, prev_state, new_state, prices_and_volumes),
+                account,
+                options,
+            )
+            .wait()
+            .map_err(DriverError::from)
+            .map(|_| ())
+    }
```

Comment:
It's not related to this PR. But some small comments about some logic here:
-  The gas limit is a bit hardcoded
-  `account_with_sufficient_balance` seams to find the first account with funds. Its fine, but I would find it ok to assume it's the first one. It seems like we foresee to set it up with a mnemonic/pks and different accounts with balances. 
- The "Not enough balance to send Txs" check only verifies that we have more than 0 ether. It should check we have enough for covering the `gas limit * gas price` we are setting
- Also, I see the API of this methods return a Result. Wouldn't it make sense to return a `Futute<Result<T>>`

Reply:
Agree with 1-3.

For the last point the question is where to put the bound for asynchronous operations. In my view it's more intuitive for the driver to interact with a synchronous interface since it needs to wait on this interaction to complete anyways (and synchronous interfaces are easier to reason about). However, if we find ourselves issuing multiple transactions which could be run in parallel I'd be fine changing it to return non blocking futures.

Reply:
Fair enough!